experimental/instrumentation-fetch/src/instrumentation.ts

```diff
@@ -0,0 +1,84 @@
+import { BaseInstrumentation, faro, globalObject } from '@grafana/faro-core';
+
+import { fetchGlobalObjectKey, originalFetch, originalFetchGlobalObjectKey } from './constants';
+
+export class FetchInstrumentation extends BaseInstrumentation {
+  readonly name = '@grafana/faro-web-sdk:instrumentation-fetch';
+  readonly version = '1.0.0'; // TODO - pull from package.json
+
+  initialize(): void {
+    console.log('Initializing FetchInstrumentation');
+    this.enable();
+  }
+
+  constructor() {
+    super();
+  }
+
+  /**
+   * Instrument fetch with Faro
+   */
+  private _instrumentFetch(): typeof fetch {
+    return function instrumentedFetch(input, init): Promise<Response> {
+      let copyInit = init ? Object.assign({}, init) : {};
+
+      let body;
+      if (copyInit && copyInit.body) {
+        body = copyInit.body;
+        copyInit.body = undefined;
+      }
+
+      const request = new Request(input, copyInit);
+      if (body) {
+        copyInit.body = body;
+      }
+
+      return originalFetch(input instanceof Request ? request : input, copyInit).then(function (response: Response) {
+        const simplifiedHeaders = {} as Record<string, string>;
+        new Headers(response.headers).forEach((v, k) => (simplifiedHeaders[`response_header_${k}`] = v));
+
+        // TODO - add instrumented functions to capture onError and onSuccess
```

Comment:
Q: Certain data like url etc. is not tracked. I think you'll add it later but want to give a hint anyways :)

Reply:
Suggestion/idea: Consider sending object with respective props like e. g.

```ts
{
    timesTamp: ...
    urls: ...
    headers: ...
    ...
}

Reply:
I love the hints thank you!

Reply:
Also, just to clarify - `timestamp` seems to be one of the fields handled in the `core` package. So providing timestamp twice doesn't seem necessary?

Reply:
> 

Sorry my bad.Yes you're absolutely right. The core package takes care of that.